src/codegen/mod.rs

```diff
@@ -1647,10 +1648,18 @@ impl<'a> FieldCodegen<'a> for Bitfield {
                 #[inline]
                 #access_spec fn #getter_name(&self) -> #bitfield_ty {
                     unsafe {
-                        ::#prefix::mem::transmute(
-                            self.#unit_field_ident.as_ref().get(#offset, #width)
-                                as #bitfield_int_ty
-                        )
+                        let val = self.#unit_field_ident.as_ref().get(
+                            #offset,
+                            #width
+                        );
+                        if (#is_signed) {
```

Comment:
No braces, please.

Instead of having this branch in the generated code, we should instead change the generated code depending on whether the signed field is signed.